augur/mask.py

```diff
@@ -1,65 +1,136 @@
 """
 Mask specified sites from a VCF file.
 """
-
-from Bio import SeqIO
-import pandas as pd
 import os
+import sys
+from shutil import copyfile
+
 import numpy as np
-from .utils import run_shell_command, shquote
+import pandas as pd
+from Bio import SeqIO
 
-def get_mask_sites(vcf_file, mask_file):
-    '''
-    Creates a temporary file in correct format for vcftools to use
-    (two-column, tab-seperated: "chromName" "position")
-    '''
+from .utils import run_shell_command, shquote, open_file, is_vcf
 
-    #Need CHROM name from VCF file:
-    import gzip
-    chromName = None
-    opn = gzip.open if vcf_file.lower().endswith('.gz') else open
-    with opn(vcf_file, mode='rt') as f: #'rt' necessary for gzip
+def get_chrom_name(vcf_file):
+    """Read the CHROM field from the first non-header line of a vcf file.
+    
+    Returns:
+    str or None: Either the CHROM field or None if no non-comment line could be found. 
+    """
+    with open_file(vcf_file, mode='r') as f:
         for line in f:
             if line[0] != "#":
-                header = line.strip().split('\t')
-                chromName = header[0]
-                break   # once chrom is found, no need to go through rest
+                header = line.strip().partition('\t')
+                return header[0]
 
-    if chromName is None or not chromName:
-        print("ERROR: Something went wrong reading your VCF file: a CHROM column could not be found. "
-              "Please check the file is valid VCF format.")
-        return None
+def read_bed_file(mask_file):
+    """Read the full list of excluded sites from the BED file.
 
-    #Read in BED file - 2nd column always chromStart, 3rd always chromEnd
-    #I timed this against sets/update/sorted; this is faster
-    sitesToMask = []
+    Second column is chromStart, 3rd is chromEnd. Generate a range from these two columns.
+    """
+    sites_to_mask = []
     bed = pd.read_csv(mask_file, sep='\t')

```

Comment:
[The basic BED format](https://genome.ucsc.edu/FAQ/FAQformat.html#format1) doesn't support an explicit header line, so this line should actually include a `header=None` argument to avoid skipping the first line. This is a very old bug as you can see from [the mask sites for TB that repeat the first site](https://github.com/nextstrain/tb/blob/master/config/drm_sites.txt#L1-L2).

Reply:
Whoops. Yeah, makes sense. Per the comment above, looks like the same bug's in `tree`, happy to fix it when we merge the two